### PR TITLE
Fix #1347 Draft file name "Unknown".

### LINKED ISSUE #1347: Draft file name "Unknown"
- Android version: 9
- Device: Fairphone 3
When sharing a file to Delta Chat and then selecting a chat, "Unknown" is shown instead of the file name. What I found out so far: The filename is still there at the beginning of `AttatchmentManager.setMedia()` (in the "uri" variable) but vanishes somewhere before `documentView.setDocument((DocumentSlide) slide);` is called. Reported at https://github.com/deltachat/deltachat-android/pull/1342.

### PR DESCRIPTION
**Pull request by Hocuri**
Fix #1347 Draft file name "Unknown".

getManuallyCalculatedSlideInfo() in AttatchmentManager sometimes failed to get a file name. Now, in this case, it is just extracted from the uri.

### CODE REVIEW

File: src/org/thoughtcrime/securesms/mms/AttachmentManager.java [hunk 1]
```diff
@@ -352,6 +353,10 @@ public void onReceivedDuration(int millis) {
           height = dimens.second;
         }
 
+        if (fileName == null) {
+          fileName = new File(uri.getPath()).getName();
```
Comment (r10s) [inline src/org/thoughtcrime/securesms/mms/AttachmentManager.java:357]: `uri.getPath()` may return null, in this case, `new File()` will throw an error.

i suggest to add a try/catch block around this statement.

Comment (Hocuri) [reply src/org/thoughtcrime/securesms/mms/AttachmentManager.java]: Ough, you're right. I'm not used to looking for this kind of problem anymore after programming Rust and Kotlin a lot :wink:

Does a try/catch block have any advantage over just checking for null?

**Comment by Hocuri** [comment]
> @Hocuri so, you could you reproduce #1347 in general?

Yes.

File: src/org/thoughtcrime/securesms/mms/AttachmentManager.java (hunk 1 above)
Comment (r10s) [reply src/org/thoughtcrime/securesms/mms/AttachmentManager.java]: in this case, checking for `null` would probably do the job - esp. as we just checked the possible return values.

but one is not always that accurate, called function may be adapted, implementations can change - and `getPath()` and `getName()` might also throw an error in the future. what, if getPath() returns a path `new File()` cannot handle?

so, assuming, that there are many things that can go wrong in uri to file conversion, i would go for a try/catch block for exactly this added line (although i am not the biggest fan of try/catch :) 

also, try/catch on that line makes sure we do not get a regression by this fix.

Comment (Hocuri) [reply src/org/thoughtcrime/securesms/mms/AttachmentManager.java]: Ok.

**Pull request merged**